src/LibYear.Lib/FileTypes/XmlProject.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using NuGet.Versioning;
+
+namespace LibYear.Lib.FileTypes
+{
+    public abstract class XmlProject : IProjectFile
+    {
+        protected readonly XDocument _xmlContents;
+        protected readonly Stream _underlyingStreamData;
+        public virtual string FileName => throw new NotSupportedException();

```

Comment:
Because the class is abstract I think `public abstract FileName {get;}` would be better. If you really _really_ want a default value of "SURPRISE HERES AN EXCEPTION" then it may be nice to also have a `public virtual bool SupportsFileName {get;}`.